ouroboros-network/src/Ouroboros/Network/PeerSelection/KnownPeers.hs

```diff
@@ -352,12 +352,18 @@ setGossipTime peeraddrs time
 -- Tracking when we can (re)connect
 --
 
-minConnectTime :: Ord peeraddr => KnownPeers peeraddr -> Maybe Time
-minConnectTime KnownPeers {
+minConnectTime :: Ord peeraddr
+               => Set peeraddr
+               -- ^ peers that are either established or which we are
+               -- establishing a connection
+               -> KnownPeers peeraddr
+               -> Maybe Time
+minConnectTime establishedPeers
+               KnownPeers {
                    availableToConnect,
                    nextConnectTimes
                  }
-  | Set.null availableToConnect
+  | Set.null (availableToConnect Set.\\ establishedPeers)
```

Comment:
Hmm. I wonder if it wouldn't be better to move this condition into the guard for `EstablishedPeers.belowTarget`. This is mixing in information about the established peers when answering a question about min connect time for known peers. They should not need to know about each other.

It's now a non-trivial condition that involves both. I think therefore it better lives in the decision guard (which has lots of such subtle guard conditions).

Reply:
Done, changes are in 9ceb0905f.